Result/ResultType.swift

```diff
@@ -101,3 +101,29 @@ infix operator &&& {
 public func &&& <L: ResultType, R: ResultType where L.Error == R.Error> (left: L, @autoclosure right: () -> R) -> Result<(L.Value, R.Value), L.Error> {
 	return left.flatMap { left in right().map { right in (left, right) } }
 }
+
+/// Return a array of `V` by applying transform to all elements of array 'U' and filtering out failure ones
+public func mapFilter<U, V, Error>(results: [U], @noescape transform: U -> Result<V, Error>) -> [V]{
```

Comment:
If expressed as an extension method on `CollectionType`, I think this becomes essentially equivalent to a straightforward use of the collection-to-optional variant of `flatMap` in the standard library:

```swift
values.mapFilter(transform)
// is equivalent to
values.flatMap { transform($0).value }
```

In which case maybe we’d be better to just use the standard library’s functionality and not extend it with this particular method.